### ISSUE TITLE
Freeze not working

### ISSUE THREAD

**Issue by SteliosStefanatos** [ipython-contrib/jupyter_contrib_nbextensions#940]
Freeze does not work properly for me. In "read-only" mode I am able to edit (and execute) the cell and in "freeze" mode I am able to execute the cell. Actually, I can still edit the cell contents in freeze mode, however, the last "unrestricted" version of the cell will be executed.

I have installed ver. 0.2.6 of `jupyter_contrib_nbextensions` and I am using Linux, Firefox, and Anaconda (python 3.6) with the latest updates (`conda update --all`). Freeze behavior is the same when I disable all other extensions.

P.S.: Thank you to all the contributors of nbextensions for your amazing work!

**Comment by jcb91** [comment]
Hmm, I'm afraid I can;t think of an obvious reason for this problem right now. Do you see any warnings or error message in the 
[javascript console](http://webmasters.stackexchange.com/questions/8525/how-to-open-the-javascript-console-in-different-browsers)? Ping @kukanya as freeze author...

**Comment by SteliosStefanatos** [comment]
Thank you for the reply.
I am getting the following javascript warnings, don't know if they are relevant or not...

```
Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead[Learn More]  main.min.js:1

Use of Mutation Events is deprecated. Use MutationObserver instead.  main.min.js:5:7259

Use of getPreventDefault() is deprecated.  Use defaultPrevented instead.  main.min.js:5:12747

unreachable code after return statement[Learn More]  highlighter.js:98:4
```

BTW, I stated in my original post that I could execute the cell in "freeze" mode, which is not the case; I cannot execute a "freezed" cell, which is the expected behavior. So the problem is only that I can edit the cell (either in "read-only" or "freeze" mode), with the edits applied when I execute in "read-only" mode.

**Comment by tnarik** [comment]
If it were modified to manipulate the 'editable' metadata property of the cell, then it would work correctly.
I'm not familiar with Jupyter Notebook extensions, so not sure how easy it would be to change that behaviour, or if it is already covered by a different extension (wasn't able to find one).

**Comment by juhasch** [comment]
The reason for this behavior is that notebook 5.0 introduces it's own implementation of read-only cells.
This extension needs to be updated. For now, it would be sufficient to replace the line
```    
 cell.code_mirror.setOption('readOnly', cell.metadata.run_control.read_only);
```
with this line
```
cell.metadata.editable = !cell.metadata.run_control.read_only;
```

**Comment by tnarik** [comment]
Yep, I modified it on my fork (plus used a cleaner `run_control` structure). It think the issue was also in 4.3.x, because there was a backport of the behaviour around `editable`.

If you want to test, @SteliosStefanatos, you can use `https://github.com/tnarik/jupyter_contrib_nbextensions/tarball/freeze_read_only`.

There are two reasons why I won't do a PR:
- It's @kukanya code AFAIK, so maybe she prefers doing it differently.
- As I said, I never played with Jupyter Notebook extensions and I'm not sure about how to indicate things like compatibility (this should be 4.x-5x, AFAIK).

But the code is there, so just shout.

**Comment by jcb91** [comment]
closing, as this should now be fixed by #968 - as always, please reopen if it turns out not to be fixed!
